src/EnergyPlus/ScheduleManager.cc

```diff
@@ -1641,354 +1636,229 @@ namespace ScheduleManager {
-                numerrors = 0;
-                rowCnt = 0;
-                if (skiprowCount > 0) {   // Numbers(2) has number of rows to skip
-                    while (!LineIn.eof) { // end of file
-                        LineIn = SchdFile.readLine();
-                        ++rowCnt;
-                        if (rowCnt == skiprowCount) {
-                            break;
+                // day schedule
+                state.dataScheduleMgr->DaySchedule(AddDaySch).Name = schedule.name + "_dy_" + ExtraField;
+                state.dataScheduleMgr->DaySchedule(AddDaySch).ScheduleTypePtr = state.dataScheduleMgr->Schedule(schedule.SchNum).ScheduleTypePtr;
+                // schedule is pointing to the week schedule
+                state.dataScheduleMgr->Schedule(schedule.SchNum).WeekSchedulePointer(iDay) = AddWeekSch;
+                int iRowNum{};
+                if (schedule.MinutesPerItem == 60) {
+                    for (int jHour = 1; jHour <= 24; ++jHour) {
+                        for (int TS = 1; TS <= state.dataGlobal->NumOfTimeStepInHour; ++TS) {
+                            state.dataScheduleMgr->DaySchedule(AddDaySch).TSValue(TS, jHour) = schedule.vals[rowCountFld];
                         }
+                        ++rowCountFld; // hourlyFileValues((hDay - 1) * 24 + jHour)
                     }
-                }
-
-                //  proper number of lines are skipped.  read the file
-                // for the rest of the lines read from the file
-                rowCnt = 0;
-                firstLine = true;
-                while (!LineIn.eof) { // end of file
-                    LineIn = SchdFile.readLine();
-                    ++rowCnt;
-                    colCnt = 0;
-                    wordStart = 0;
-                    columnValue = 0.0;
-                    // scan through the line looking for a specific column
-                    while (true) {
-                        sepPos = index(LineIn.data, ColumnSep);
-                        ++colCnt;
-                        if (sepPos != std::string::npos) {
-                            if (sepPos > 0) {
-                                wordEnd = sepPos - 1;
-                            } else {
-                                wordEnd = wordStart;
+                } else { // Minutes Per Item < 60
+                    if (schedule.FileIntervalInterpolated) {
+                        for (int Hr = 1; Hr <= 24; ++Hr) {
+                            int minuteEnd = schedule.MinutesPerItem;
+                            SCount = 1;
+                            CurMinute = state.dataGlobal->MinutesPerTimeStep;
+                            for (iRowNum = 1; iRowNum <= hrLimitCount; ++iRowNum) {
+                                MinuteValue({SCount, minuteEnd}, Hr) = schedule.vals[rowCountFld];
+                                SCount = minuteEnd + 1;
+                                minuteEnd += schedule.MinutesPerItem;
+                                ++rowCountFld;
```

Comment:
Same as the previous code, except that I combined the loops that were iterating between for (int Hr = 1; Hr <= 24; ++Hr).